esx_service/utils/kvESX.py

```diff
@@ -166,35 +168,45 @@ def delete(volpath):
     return True
 
 
+# Align a given string to the specified block boundary.
+def align_str(kv_str, block):
+   # Align string to the next block boundary.
+   new_len = int((len(kv_str) + block - 1) / block) * block - 1
+   padded_str = '{:<{width}}'.format(kv_str, width=new_len)
+
+   # Add a trailing newline to bring up string length to a 4K boundary.
+   return '{0}{1}'.format(padded_str, '\n')
+
 # Load and return dictionary from the sidecar
 def load(volpath):
     metaFile = lib.DiskLib_SidecarMakeFileName(volpath, DVOL_KEY)
 
     try:
-        fh = open(metaFile, "r+")
-    except IOError:
-        logging.exception("Failed to open %s", metaFile);
+       with open(metaFile, "r+") as fh:
```

Comment:
why 'r+', not 'r' ? we don't see to be writing anything, so 'r' should work, should not it ?